### PR TITLE
Add tests for reading back contents from WebGPU canvas

### PR DESCRIPTION
**Pull request by shaoboyan**
Add test cases to cover WebGPU canvas to do snapshot
or interact with WebGL.

<hr>

**Author checklist for test code/plans:**

- [x] All outstanding work is tracked with "TODO" in a test/file description or `.unimplemented()` on a test.
- [x] New helpers, if any, are documented using `/** doc comments */` and can be found via `helper_index.txt`.
- [x] (Optional, sometimes not possible.) Tests pass (or partially pass without unexpected issues) in an implementation. (Add any extra details above.)

**[Reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md) for test code/plans:** (Note: feel free to pull in other reviewers at any time for any reason.)

- [ ] The test path is reasonable, the [description](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) "describes the test, succinctly, but in enough detail that a reader can read only the test plans in a file or directory and evaluate the completeness of the test coverage."
- [ ] Tests appear to cover this area completely, except for outstanding TODOs. Validation tests use control cases.
    (This is critical for coverage. Assume anything without a TODO will be forgotten about forever.)
- [ ] Existing (or new) test helpers are used where they would reduce complexity.
- [ ] TypeScript code is readable and understandable (is unobtrusive, has reasonable type-safety/verbosity/dynamicity).

### CODE REVIEW

**Comment by shaoboyan** [comment]
@kainino0x  PTAL, thanks!

File: src/webgpu/web_platform/canvas/readbackFromWebGPUCanvas.spec.ts [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+    gl.bindFramebuffer(gl.FRAMEBUFFER, fb);
+    gl.framebufferTexture2D(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.TEXTURE_2D, texture, 0);
+
+    const pixels = new Uint8Array(canvas.width * canvas.height * 4);
+    gl.readPixels(0, 0, 2, 2, gl.RGBA, gl.UNSIGNED_BYTE, pixels);
+    const actual = new Uint8ClampedArray(pixels);
+
+    t.expectOK(checkElementsEqual(actual, webglExpect));
+  });
```
Comment (kainino0x) [inline src/webgpu/web_platform/canvas/readbackFromWebGPUCanvas.spec.ts:265]: We should have one more test for using a webgpu canvas as a source to 2d canvas `drawImage`

**Review by kainino0x — APPROVED** [review]
Looks great! One comment to add one more test case - can be a revision to this PR, or a new PR. Up to you.

File: src/webgpu/web_platform/canvas/readbackFromWebGPUCanvas.spec.ts (hunk 1 above)
Comment (shaoboyan) [reply src/webgpu/web_platform/canvas/readbackFromWebGPUCanvas.spec.ts:278]: Ah, missing this. Added, thanks!

**Comment by shaoboyan** [comment]
@kainino0x  Please take another look for new adding 2d drawImage case. Thanks!

**Review by kainino0x — APPROVED** [review]

**Pull request merged**